cmd/storagenode-updater/main.go

```diff
@@ -200,23 +234,60 @@ func update(ctx context.Context, nodeID storj.NodeID) (err error) {
 				return errs.New("invalid version downloaded: wants %s got %s", suggestedVersion.String(), downloadedVersion.String())
 			}
 
-			log.Println("restarting service", runCfg.ServiceName)
-			err = restartSNService(runCfg.ServiceName)
+			log.Println("restarting service", serviceName)
+			err = restartService(serviceName)
 			if err != nil {
 				// TODO: should we try to recover from this?
 				return errs.New("unable to restart service: %v", err)
 			}
-			log.Println("service", runCfg.ServiceName, "restarted successfully")
+			log.Println("service", serviceName, "restarted successfully")
 
 			// TODO remove old binary ??
 		} else {
-			log.Printf("%s version is up to date\n", runCfg.ServiceName)
+			log.Printf("%s version is up to date\n", serviceName)
 		}
 	}
 
 	return nil
 }
 
+func renameStoragenode(currentVersion version.SemVer) error {
```

Comment:
maybe instead of two almost identical functions something like this:
```
func renameBinary(binaryLocation, backupName string) error {
	extension := filepath.Ext(binaryLocation)
	if extension != "" {
		extension = "." + extension
	}

	dir := filepath.Dir(binaryLocation)
	backupExec := filepath.Join(dir, backupName)

	if err := os.Rename(binaryLocation, backupExec); err != nil {
		return errs.Wrap(err)
	}
	return nil
}
```